internal/csi-addons/rbd/replication.go

```diff
@@ -937,3 +947,35 @@ func checkVolumeResyncStatus(ctx context.Context, localStatus types.SiteStatus)
 
 	return nil
 }
+
+func getCurrentReplicationStatus(mirrorGlobalStatus types.GlobalStatus,
+) (replication.GetVolumeReplicationInfoResponse_Status, string, error) {
+	// get image local status
+	localStatus, err := mirrorGlobalStatus.GetLocalSiteStatus()
+	if err != nil {
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "", err
+	}
+
+	// parse only the message from the description of the image/group status
+	statusDesc := localStatus.GetDescription()
+	desc := strings.SplitN(statusDesc, ",", 2)[0]
+
+	// check if mirroring state is up/down
+	isUp := localStatus.IsUP()
+
+	if !isUp {
+		return replication.GetVolumeReplicationInfoResponse_DEGRADED, desc, nil
+	}
+
+	resp := replication.GetVolumeReplicationInfoResponse_UNKNOWN
+	switch localStatus.GetState() {
+	case librbd.MirrorImageStatusStateError.String():
+		resp = replication.GetVolumeReplicationInfoResponse_ERROR
+	case librbd.MirrorImageStatusStateReplaying.String():
+		fallthrough
+	case librbd.MirrorImageStatusStateStopped.String():
+		resp = replication.GetVolumeReplicationInfoResponse_HEALTHY
+	}
```

Comment:
add a default case and log it, This helps for debugging in future?

Reply:
now make sense to log it for debugging

Reply:
Ack, added a log.